accord-core/src/main/java/accord/impl/InMemoryCommandStore.java

```diff
@@ -576,15 +576,25 @@ public <T> Future<T> submit(PreLoadContext context, Function<? super SafeCommand
 
         public SingleThread(int id, NodeTimeService time, Agent agent, DataStore store, ProgressLog.Factory progressLogFactory, RangesForEpochHolder rangesForEpoch)
         {
-            super(id);
-            executor = Executors.newSingleThreadExecutor(r -> {
+            this(id, time, agent, store, progressLogFactory, rangesForEpoch, Executors.newSingleThreadExecutor(r -> {
                 Thread thread = new Thread(r);
                 thread.setName(CommandStore.class.getSimpleName() + '[' + time.id() + ']');
                 return thread;
-            });
+            }));
+        }
+
+        private SingleThread(int id, NodeTimeService time, Agent agent, DataStore store, ProgressLog.Factory progressLogFactory, RangesForEpochHolder rangesForEpoch, ExecutorService executor)
```

Comment:
name may not make sense in this case, 100% cool making a base type such as `ExecutorCommandStore` and have `SingleThread` extend it and use the existing logic to build the single threaded executor